zappa/core.py

```diff
@@ -1483,34 +1487,18 @@ def rollback_lambda_function_version(
 
         return response["FunctionArn"]
 
-    def is_lambda_function_ready(self, function_name):
-        """
-        Checks if a lambda function is active and no updates are in progress.
-        """
-        response = self.lambda_client.get_function(FunctionName=function_name)
-        return (
-            response["Configuration"]["State"] == "Active"
-            and response["Configuration"]["LastUpdateStatus"] != "InProgress"
-        )
-
-    def wait_until_lambda_function_is_ready(self, function_name):
+    def wait_until_lambda_function_is_ready(self, lambda_arn):
         """
         Continuously check if a lambda function is active.
         For functions deployed with a docker image instead of a
         ZIP package, the function can take a few seconds longer
         to be created or update, so we must wait before running any status
         checks against the function.
```

Comment:
May be worth updating this docstring? Originally I added it because we needed to wait until the function was ready for docker deployments, but it sounds like there are other use cases so maybe we should make it more general?